include/opw_kinematics/opw_kinematics_impl.h

```diff
@@ -132,69 +133,25 @@ Solutions<T> inverse(const Parameters<T>& params, const Transform<T>& pose) noex
-  out3[5] = (theta6_iv + params.offsets[5]) * params.sign_corrections[5];
-
-  Solution<T>& out4 = solutions[4];
-  out4[0] = (theta1_i + params.offsets[0]) * params.sign_corrections[0];
-  out4[1] = (theta2_i + params.offsets[1]) * params.sign_corrections[1];
-  out4[2] = (theta3_i + params.offsets[2]) * params.sign_corrections[2];
-  out4[3] = (theta4_v + params.offsets[3]) * params.sign_corrections[3];
-  out4[4] = (theta5_v + params.offsets[4]) * params.sign_corrections[4];
-  out4[5] = (theta6_v + params.offsets[5]) * params.sign_corrections[5];
-
-  Solution<T>& out5 = solutions[5];
-  out5[0] = (theta1_i + params.offsets[0]) * params.sign_corrections[0];
-  out5[1] = (theta2_ii + params.offsets[1]) * params.sign_corrections[1];
-  out5[2] = (theta3_ii + params.offsets[2]) * params.sign_corrections[2];
-  out5[3] = (theta4_vi + params.offsets[3]) * params.sign_corrections[3];
-  out5[4] = (theta5_vi + params.offsets[4]) * params.sign_corrections[4];
-  out5[5] = (theta6_vi + params.offsets[5]) * params.sign_corrections[5];
-
-  Solution<T>& out6 = solutions[6];
-  out6[0] = (theta1_ii + params.offsets[0]) * params.sign_corrections[0];
-  out6[1] = (theta2_iii + params.offsets[1]) * params.sign_corrections[1];
-  out6[2] = (theta3_iii + params.offsets[2]) * params.sign_corrections[2];
-  out6[3] = (theta4_vii + params.offsets[3]) * params.sign_corrections[3];
-  out6[4] = (theta5_vii + params.offsets[4]) * params.sign_corrections[4];
-  out6[5] = (theta6_vii + params.offsets[5]) * params.sign_corrections[5];
-
-  Solution<T>& out7 = solutions[7];
-  out7[0] = (theta1_ii + params.offsets[0]) * params.sign_corrections[0];
-  out7[1] = (theta2_iv + params.offsets[1]) * params.sign_corrections[1];
-  out7[2] = (theta3_iv + params.offsets[2]) * params.sign_corrections[2];
-  out7[3] = (theta4_viii + params.offsets[3]) * params.sign_corrections[3];
-  out7[4] = (theta5_viii + params.offsets[4]) * params.sign_corrections[4];
-  out7[5] = (theta6_viii + params.offsets[5]) * params.sign_corrections[5];
+  Eigen::Map<const Eigen::Array<T, 6, 1>> offsets(params.offsets.data());
+  Eigen::Array<T, 6, 1> signs;
+  signs[0] = params.sign_corrections[0];
+  signs[1] = params.sign_corrections[1];
+  signs[2] = params.sign_corrections[2];
+  signs[3] = params.sign_corrections[3];
+  signs[4] = params.sign_corrections[4];
+  signs[5] = params.sign_corrections[5];
+
+  Eigen::Array<T, 6, 8> theta;
+  theta << theta1_i, theta1_i, theta1_ii, theta1_ii, theta1_i, theta1_i, theta1_ii, theta1_ii, theta2_i, theta2_ii,
+      theta2_iii, theta2_iv, theta2_i, theta2_ii, theta2_iii, theta2_iv, theta3_i, theta3_ii, theta3_iii, theta3_iv,
+      theta3_i, theta3_ii, theta3_iii, theta3_iv, theta4_i, theta4_ii, theta4_iii, theta4_iv, theta4_v, theta4_vi,
+      theta4_vii, theta4_viii, theta5_i, theta5_ii, theta5_iii, theta5_iv, theta5_v, theta5_vi, theta5_vii, theta5_viii,
+      theta6_i, theta6_ii, theta6_iii, theta6_iv, theta6_v, theta6_vi, theta6_vii, theta6_viii;
```

Comment:
I would suggest changing this to reflect the `theta` values for each individual solution on a line-by-line basis. If we ever read back through this code to compare to the paper, it's probably going to be really difficult to understand what we're doing here, especially because the theta values are not consistently different between solutions. My suggestion:

```c++
// clang-format off
theta << 0, 0, 0, 0, 0, 0,   // theta values for solution 0
  1, 1, 1, 1, 1, 1,  // theta values for solution 1
  ...
  7, 7, 7, 7, 7, 7;  // theta values for solution 7
// clang-format on
```
I think you could do this in two ways:

1. Use row-major arrays with ` Eigen::RowMajorBit` as the fourth template parameter and call `rowwise` instead of `colwise` below, or:
1. Stream the values in row-major format (which is more intuitive to read) and then transpose this array to make the math work correctly. If you take this approach, I would add a comment explaining this for posterity

Reply:
I disabled the clang format and cleaned up what it did so it is easier to read.